galley/pkg/config/analysis/analyzers/auth/mtls/policy_checker.go

```diff
@@ -73,52 +76,82 @@ func (w TargetService) PortName() string {
 // TargetService will require MTLS when an incoming connection occurs using the
 // IsServiceMTLSEnforced() call.
 type PolicyChecker struct {
-	// meshHasStrictMTLSPolicy tracks whether or not mTLS is strictly enforced on the mesh.
-	meshHasStrictMTLSPolicy bool
+	meshMTLSModeAndResource ModeAndResource
 
-	namespaceToMTLSMode map[string]strictMTLSMode
-	serviceToMTLSMode   map[TargetService]strictMTLSMode
+	namespaceToMTLSMode map[string]ModeAndResource
+	serviceToMTLSMode   map[TargetService]ModeAndResource
 }
 
-// strictMTLSMode is a helper type used to represent whether or not MTLS was
-// explicitly enabled, explicitly disabled, or just not present. Useful for the
-// IsServiceMTLSEnforced check to determine which policy applies.
-type strictMTLSMode int
+// Mode is a special type used to distinguish between MTLS being off
+// (unsupported), permissive (supported but not required), and strict (required).
```

Comment:
if by `off` you mean plaintext, that can be configured, by creating a policy without `peers`.

Reply:
Renamed to 'Plaintext'.